analyzers/tests/SonarAnalyzer.UnitTest/Metrics/CSharpExecutableLinesMetricTest.cs

```diff
@@ -627,6 +627,205 @@ public void Foo(bool flag)
+                        {
+                            for         // +1
+                              (         // +0
+                               int i=0; // +0
+                               i < 10;  // +0
+                               i++      // +0
+                              )         // +0
+                            {           // +0
+                            }
+                        }",
+              4);
+
+        [TestMethod]
+        public void SwitchStatementWithMultipleCases() =>
+            AssertLinesOfCode(
+              @"
+                        void Foo(int? i, string s)
+                        {
+                            switch (i) // +1
+                            {
+                                case 1:
+                                    Console.WriteLine(4); // +1
+                                    break; // +1
+                                case 2:
+                                    Console.WriteLine(4); // +1
+                                    break; // +1
+                                default:
+                                    break; // +1
+                            }
+                        }",
+              4, 7, 8, 10, 11, 13);
+
+        [TestMethod]
+        public void SwitchExpressionWithMultipleCases() =>
+            AssertLinesOfCode(
+              @"
+                        void Foo(int? i, string s)
+                        {
+                            var x = s switch
+                            {
+                                    ""a"" => true, // FN
+                                    ""b"" => false, // FN
+                                    _ => true // FN
+                            };
+                            var y = s switch
+                            {
+                                ""a"" => Foo(""b""), // +1
+                                ""b"" => Foo(""a""), // +1
+                                _ => false // FN
+                            };
+                        }
+                        bool Foo(string s) => true;
+                ",
+              12, 13);
+
+        [TestMethod]
+        public void MultiLineInterpolatedString() =>
+            AssertLinesOfCode(
+              @"
+                        void Foo(int? i, string s)
+                        {
+                            string x = ""someString"";
+                            x += @$""This is a Multi
+                                                  Line
+                                                  interpolated
+                                                  string {i} {s}"";
+                        }
+                ",
+              5);
+
+        [TestMethod]
+        public void UsingDeclaration() =>
+            AssertLinesOfCode(
+              @"
+                        void Foo(int? i, string s)
+                        {
+                            using var file = new System.IO.StreamWriter(""WriteLines2.txt""); // FN
```

Comment:
You could rephrase  #4689 to mention all what is not covered (not only expression bodied methods, but also using statements etc).